flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/UnsignedTypeConversionITCase.java

```diff
@@ -94,70 +103,76 @@
-            Assertions.assertThat(resultSet.getObject("tiny_un_c")).isEqualTo(255);
-            Assertions.assertThat(resultSet.getObject("small_c")).isEqualTo(32767);
-            Assertions.assertThat(resultSet.getObject("small_un_c")).isEqualTo(65535);
-            Assertions.assertThat(resultSet.getObject("int_c")).isEqualTo(2147483647);
-            Assertions.assertThat(resultSet.getObject("int_un_c")).isEqualTo(4294967295L);
-            Assertions.assertThat(resultSet.getObject("big_c")).isEqualTo(9223372036854775807L);
-            Assertions.assertThat(resultSet.getObject("big_un_c"))
-                    .isEqualTo(new BigInteger("18446744073709551615"));
+        try (Connection con =
+                        DriverManager.getConnection(MYSQL_CONTAINER.getJdbcUrl(), USER, PASSWORD);
+                PreparedStatement ps =
+                        con.prepareStatement(
+                                String.format(
+                                        "select %s from %s",
+                                        String.join(",", COLUMNS), TABLE_NAME))) {
+            ResultSet resultSet = ps.executeQuery();
+            while (resultSet.next()) {
+                assertThat(resultSet.getObject("tiny_c")).isEqualTo(127);
+                assertThat(resultSet.getObject("tiny_un_c")).isEqualTo(255);
+                assertThat(resultSet.getObject("small_c")).isEqualTo(32767);
+                assertThat(resultSet.getObject("small_un_c")).isEqualTo(65535);
+                assertThat(resultSet.getObject("int_c")).isEqualTo(2147483647);
+                assertThat(resultSet.getObject("int_un_c")).isEqualTo(4294967295L);
+                assertThat(resultSet.getObject("big_c")).isEqualTo(9223372036854775807L);
+                assertThat(resultSet.getObject("big_un_c"))
+                        .isEqualTo(new BigInteger("18446744073709551615"));
+            }
         }
 
         // read data from db using flink and compare
         Iterator<Row> collected =
-                tEnv.executeSql(String.format("select %s from jdbc_source", COLUMNS)).collect();
+                tEnv.executeSql(
+                                String.format(
+                                        "select %s from jdbc_source", String.join(",", COLUMNS)))
+                        .collect();
         List<Row> result = CollectionUtil.iteratorToList(collected);
-        List<Row> expected = Collections.singletonList(Row.ofKind(RowKind.INSERT, ROW));
-        Assertions.assertThat(result).isEqualTo(expected);
-
-        connection.close();
+        assertThat(result).containsOnly(Row.ofKind(RowKind.INSERT, ROW));
     }
 
     private void prepare() throws Exception {
-        MYSQL_CONTAINER.start();
-
-        connection = DriverManager.getConnection(MYSQL_CONTAINER.getJdbcUrl(), USER, PASSWORD);
         tEnv = StreamTableEnvironment.create(StreamExecutionEnvironment.getExecutionEnvironment());
```

Comment:
Can you also make this a local variable? It's always better in tests to have methods with parameters, rather than having test class fields. This is also in order to simplify future JUnit 5 test porting